qa/vagrant/src/main/java/org/elasticsearch/packaging/util/Shell.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.packaging.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Wrapper to run shell commands and collect their outputs in a less verbose way
+ */
+public class Shell {
+
+    final Map<String, String> env;
+    final Path workingDirectory;
+
+    public Shell() {
+        this(emptyMap(), null);
+    }
+
+    public Shell(Map<String, String> env) {
+        this(env, null);
+    }
+
+    public Shell(Path workingDirectory) {
+        this(emptyMap(), workingDirectory);
+    }
+
+    public Shell(Map<String, String> env, Path workingDirectory) {
+        this.env = new HashMap<>(env);
+        this.workingDirectory = workingDirectory;
+    }
+
+    public Result runAndAssertSuccess(String... command) {
```

Comment:
I think it is far more often we want to assert success. Can we flip this around have have bare `run` assert, and have a `runIgnoreFailure` for the other case?

Reply:
Yeah good point, it's also probably more intuitive that way